### PR TITLE
docs(runtime) Add some better runtime docs

### PR DESCRIPTION
**Pull request by lachlansneff**

### CODE REVIEW

File: lib/runtime-core/src/backing.rs [hunk 1]
```diff
@@ -17,30 +17,31 @@ use crate::{
 };
 use std::slice;
 
+/// The `LocalBacking` "owns" the memory used by all the local resources.
+/// That is, local memories, tables, and globals (as well as some additional
+/// data for the virtual call machinery).
```
Comment (bjfish) [inline lib/runtime-core/src/backing.rs]: One suggestion
> The `LocalBacking` "owns" the state used by

Could you clarify if LocalBacking belongs to an instance or module?

File: lib/runtime-core/src/backing.rs [hunk 2]
```diff
@@ -102,6 +103,9 @@ impl LocalBacking {
         memories.into_boxed_map()
     }
 
+    /// Initialize each local memory.
```
Comment (bjfish) [inline lib/runtime-core/src/backing.rs]: > Initializes each

Does `local` mean not internal or something else? Could we clarify what local is local to?

File: lib/runtime-core/src/backing.rs [hunk 3]
```diff
@@ -174,6 +178,9 @@ impl LocalBacking {
         tables.into_boxed_map()
     }
 
+    /// This initializes all of the local tables, e.g.
```
Comment (bjfish) [inline lib/runtime-core/src/backing.rs]: > Initializes all of the local tables
Consistency with above

File: lib/runtime-core/src/vm.rs [hunk 4]
```diff
@@ -9,7 +9,16 @@ use std::{ffi::c_void, mem, ptr};
 
 /// The context of the currently running WebAssembly instance.
 ///
+/// This is implicitly passed to every webassembly function.
```
Comment (bjfish) [inline lib/runtime-core/src/vm.rs]: WebAssembly

File: lib/runtime-core/src/vm.rs [hunk 5]
```diff
@@ -9,7 +9,16 @@ use std::{ffi::c_void, mem, ptr};
...
+/// Since this is per-module, each field has a statically
```
Comment (bjfish) [inline lib/runtime-core/src/vm.rs]: Should this be `per-instance`?

**Comment by Hywan** [comment]
Thanks for the PR!

**Pull request merged**